catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java

```diff
@@ -812,15 +812,16 @@ private void initScreenMode() {
 	}
 
 	private void disposeTextures() {
-		List<Sprite> sprites = scene.getSpriteList();
-		int spriteSize = sprites.size();
-		for (int i = 0; i > spriteSize; i++) {
-			List<LookData> data = sprites.get(i).getLookDataList();
-			int dataSize = data.size();
-			for (int j = 0; j < dataSize; j++) {
-				LookData lookData = data.get(j);
-				lookData.getPixmap().dispose();
-				lookData.getTextureRegion().getTexture().dispose();
+		for (Scene scene : project.getSceneList()) {
+			List<Sprite> sprites = scene.getSpriteList();
+			for (Sprite sprite : sprites) {
+				List<LookData> lookDatas = sprite.getLookDataList();
+				int lookDatasSize = lookDatas.size();
+				for (int j = 0; j < lookDatasSize; j++) {
+					LookData lookData = lookDatas.get(j);
+					lookData.getPixmap().dispose();
+					lookData.getTextureRegion().getTexture().dispose();
+				}
```

Comment:
Improve the lookDatas loop by using for-each.
Inline all variables. There is no need to declare `List<?>`